docs/PREACT_BADGES.md

```diff
@@ -0,0 +1,328 @@
+
+For example, if the locations.json file contains the following location definition:
+
+```json
+{
+	"left": [
+		{
+			"tag": "left",
+			"name": "Top Left"
+		},
+		{
+			"tag": "left-bottom",
+			"name": "Bottom Left"
+		}
+	]
+}
+```
+
+To restrict a badge template to a custom location, the badge template `locations` array should contain the `tag` of the locations. Ie. `left/left-bottom`
+
+
+### Badge Template Parameters
+Badge template parameters is an array of objects. Each object is a template parameter and contains the following properties: 
+
+#### The following badge template parameters properties are **required**:
+
+`name` - unique badge location identifier
+
+`type` - parameter value type. Available types: `array`, `string`, `color`, `url`, `integer`, `decimal`, `boolean`, `checkbox`, `toggle`. See example below for example usage of each type. 
+
+`label` - label that is displayed when selecting this badge location within the Searchspring Management Console
+
+`description` - badge location description
+
+`options` - required only if `type` is `array`. Defines a list of available values to select.
+
+
+#### The following badge template parameters properties are **optional**:
+
+`defaultValue` - default value that will be used unless specified when configuring a new badge rule. Must be a string regardless of different `type` options.
+
+`validations` - only applicable if `type` is `string`, `url`. `integer`, `decimal`
+
+`validations.min` - only applicable if `type` is `integer` or `decimal`. Ensures `defaultValue` or the user defined `value` meets a numerical minimum or `string` / `url` length
+
+`validations.max` - only applicable if `type` is `integer` or `decimal`. Ensures `defaultValue` or the user defined `value` meets a numerical maximum or `string` / `url` length
+
+`validations.regex` - ensures `defaultValue` or the user defined `value` meets a regex definition. Must also provide `validations.regexExplain`
+
+`validations.regexExplain` - required if using `validations.regex`. Describes the regex definition and is displayed as an error message if the regex validation fails
+
+
+```json
+{
+	"parameters": [
+		{
+			"name": "font_size",
+			"type": "array",
+			"label": "Font Size",
+			"description": "Select the badge font size",
+			"defaultValue": "14px",
+			"options": ["14px", "16px", "18px", "20px", "22px", "24px", "26px", "28px", "30px"]
+		},
+		{
+			"name": "prefix",
+			"type": "string",
+			"label": "Prefix Symbol",
+			"description": "Display a prefix before the badge value. Ie. currency symbol",
+			"validations": {
+				"regex": "^[$€]$",
+				"regexExplain": "Only $ or € currency symbols are allowed"
+			}
+		},
+		{
+			"name": "background",
+			"type": "color",
+			"label": "Badge background color",
+			"description": "Select the badge background color",
+			"defaultValue": "#0000ff"
```

Comment:
Since the color picker converts to rgba, should we use rgba in our docs as a best practice?